common/lib/xmodule/xmodule/tabs.py

```diff
@@ -168,610 +161,33 @@ def from_json(tab_dict):
-
-
-class WikiTab(HideableTab):
-    """
-    A tab_dict containing the course wiki.
-    """
-
-    type = 'wiki'
-
-    def __init__(self, tab_dict=None):
-        super(WikiTab, self).__init__(
-            # Translators: "Wiki" is the name of the course's wiki page
-            name=tab_dict['name'] if tab_dict else _('Wiki'),
-            tab_id=self.type,
-            link_func=link_reverse_func('course_wiki'),
-            tab_dict=tab_dict,
-        )
-
-    def can_display(self, course, settings, is_user_authenticated, is_user_staff, is_user_enrolled):
-        return settings.WIKI_ENABLED and (
-            course.allow_public_wiki_access or is_user_enrolled or is_user_staff
-        )
-
-    @classmethod
-    def validate(cls, tab_dict, raise_error=True):
-        return super(WikiTab, cls).validate(tab_dict, raise_error) and need_name(tab_dict, raise_error)
-
-
-class DiscussionTab(EnrolledOrStaffTab):
-    """
-    A tab only for the new Berkeley discussion forums.
-    """
-
-    type = 'discussion'
-
-    def __init__(self, tab_dict=None):
-        super(DiscussionTab, self).__init__(
-            # Translators: "Discussion" is the title of the course forum page
-            name=tab_dict['name'] if tab_dict else _('Discussion'),
-            tab_id=self.type,
-            link_func=link_reverse_func('django_comment_client.forum.views.forum_form_discussion'),
-        )
-
-    def can_display(self, course, settings, is_user_authenticated, is_user_staff, is_user_enrolled):
-        if settings.FEATURES.get('CUSTOM_COURSES_EDX', False):
-            from ccx.overrides import get_current_ccx  # pylint: disable=import-error
-            if get_current_ccx():
-                return False
-        super_can_display = super(DiscussionTab, self).can_display(
-            course, settings, is_user_authenticated, is_user_staff, is_user_enrolled
-        )
-        return settings.FEATURES.get('ENABLE_DISCUSSION_SERVICE') and super_can_display
-
-    @classmethod
-    def validate(cls, tab_dict, raise_error=True):
-        return super(DiscussionTab, cls).validate(tab_dict, raise_error) and need_name(tab_dict, raise_error)
-
-
-class LinkTab(CourseTab):
-    """
-    Abstract class for tabs that contain external links.
-    """
-    link_value = ''
-
-    def __init__(self, name, tab_id, link_value):
-        self.link_value = link_value
-        super(LinkTab, self).__init__(
-            name=name,
-            tab_id=tab_id,
-            link_func=link_value_func(self.link_value),
-        )
-
-    def __getitem__(self, key):
-        if key == 'link':
-            return self.link_value
-        else:
-            return super(LinkTab, self).__getitem__(key)
-
-    def __setitem__(self, key, value):
-        if key == 'link':
-            self.link_value = value
+            return None
+        tab_type.validate(tab_dict)
+        if issubclass(tab_type, CourseViewType):
```

Comment:
Do we still need this test? I think that all tab types are now ``CourseViewType``s, right?